frontend/context/StakingContractDetailsProvider.tsx

```diff
@@ -91,25 +94,48 @@ const useStakingContractDetailsByStakingProgram = ({
       stakingProgramId!,
     ),
     queryFn: async () => {
-      if (isNil(serviceNftTokenId))
-        return serviceApi.getStakingContractDetailsByStakingProgramId(
-          stakingProgramId!,
-          evmHomeChainId,
+      /**
+       * Request staking contract details
+       * if service is present, request it's info and states on the staking contract
+       */
+      const promises: Promise<
+        StakingContractDetails | StakingContactServiceInfo
+      >[] = [
+        serviceApi.getStakingContractDetails(stakingProgramId!, evmHomeChainId),
+      ];
+
+      if (!isNil(serviceNftTokenId)) {
+        promises.push(
+          serviceApi.getServiceInfo(
+            serviceNftTokenId,
+            stakingProgramId!,
+            evmHomeChainId,
+          ),
         );
-      return serviceApi.getStakingContractDetailsByServiceIdStakingProgram(
-        serviceNftTokenId!,
-        stakingProgramId!,
-        evmHomeChainId,
-      );
+      }
+
+      return Promise.allSettled(promises).then((results) => {
```

Comment:
fixes this part was undefined 

<img width="686" alt="image" src="https://github.com/user-attachments/assets/48435107-4f36-4c2e-9986-3d04c6522c67">